src/consensus/blockchain.py

```diff
@@ -501,3 +509,70 @@ async def validate_unfinished_block(
             return None, error_code
 
         return required_iters, None
+
+    def sub_block_record(self, header_hash: bytes32) -> SubBlockRecord:
+        sub_block = self.__sub_blocks[header_hash]
+        assert sub_block is not None
```

Comment:
this assert is not necessary, you can do return self.__sub_blocks[header_hash] which will throw a KeyError (as opposed to manually raising AssertionError)

Reply:
Otherwise we can return optional from this, if this is called in a situation where the block might not be present